### PR TITLE
remove some more deprecated API warnings

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Removed some other deprecation warnings popping out in `CMSSW_12_4_CMSDEPRECATED_X_2022-05-04-2300` (I guess after https://github.com/cms-sw/cmssw/pull/37592 was merged).
Technical, part of the migrations in https://github.com/cms-sw/cmssw/issues/36404.
Profited to clean-up headers and merging plugin header with implementations.

#### PR validation:

`cmssw` compiles, no warnings seen after merging.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by jpata** [comment]
@cmsbuild please test

**Comment by emanueleusai** [comment]
+1

**Comment by jpata** [comment]
+reconstruction
- technical

File: RecoPixelVertexing/PixelTrackFitting/test/PixelTrackTest.cc [hunk 1]
```diff
@@ -58,10 +58,10 @@ void PixelTrackTest::analyze(const edm::Event& ev, const edm::EventSetup& es) {
   using namespace std;
   using namespace reco;
 
-  cout << "*** PixelTrackTest, analyze event: " << ev.id() << endl;
+  edm::LogPrint("PixelTrackTest") << "*** PixelTrackTest, analyze event: " << ev.id();
   Handle<SimTrackContainer> simTrks;
   ev.getByLabel("g4SimHits", simTrks);
-  //  std::cout << "simtrks " << simTrks->size() << std::endl;
+  //  std::edm::LogPrint("PixelTrackTest") << "simtrks " << simTrks->size() << std::endl;
```
Comment (perrotta) [inline RecoPixelVertexing/PixelTrackFitting/test/PixelTrackTest.cc:64]:
```suggestion
  //  edm::LogPrint("PixelTrackTest") << "simtrks " << simTrks->size();
```
(But ok, it is in a commented out line)

**Comment by perrotta** [comment]
+1

**Pull request merged**